coretex/cli/modules/docker.py

```diff
@@ -0,0 +1,111 @@
+    # of 0, indicating success, the function returns True, meaning the network exists.
+    # If the command exits with a non-zero return code, indicating failure,
+    # the function returns False, meaning the network doesn't exist.
+    try:
+        command(["docker", "network", "inspect", name], ignoreStdout = True, ignoreStderr = True)
+        return True
+    except:
+        return False
+
+
+def containerExists(name: str) -> bool:
+    try:
+        _, output, _ = command(["docker", "ps", "--format", "{{.Names}}"], ignoreStderr = True, ignoreStdout = True)
+        return name in output
+    except:
+        return False
+
+
+def createNetwork(name: str) -> None:
+    if networkExists(name):
+        removeNetwork(name)
+
+    command(["docker", "network", "create", "--driver", "bridge", name], ignoreStdout = True)
+
+
+def removeNetwork(name: str) -> None:
+    command(["docker", "network", "rm", name], ignoreStdout = True, ignoreStderr = True)
+
+
+def imagePull(image: str) -> None:
+    command(["docker", "image", "pull", image], ignoreStdout = True)
+
+
+def start(
+    name: str,
+    dockerImage: str,
+    imageType: str,
+    serverUrl: str,
+    storagePath: str,
+    nodeAccessToken: str,
+    nodeRam: int,
+    nodeSwap: int,
+    nodeSharedMemory: int,
+) -> None:
+
+    runCommand = [
+        "docker", "run", "-d",
+        "--env", f"CTX_API_URL={serverUrl}",
+        "--env", f"CTX_STORAGE_PATH={storagePath}",
+        "--env", f"CTX_NODE_ACCESS_TOKEN={nodeAccessToken}",
+        "--restart", 'always',
+        "-p", "21000:21000",
+        "--cap-add", "SYS_PTRACE",
+        "--network", name,
+        "--memory", f"{nodeRam}G",
+        "--memory-swap", f"{nodeSwap}G",
+        "--shm-size", f"{nodeSharedMemory}G",
+        "--name", name,
+    ]
+
+    if imageType == "gpu":
+        runCommand.extend(["--gpus", "all"])
+
+    runCommand.append(dockerImage)
+    command(runCommand, ignoreStdout = True)
+
+
+def stopContainer(name: str) -> None:
+    command(["docker", "stop", name], ignoreStdout = True, ignoreStderr = True)
+    command(["docker", "rm", name], ignoreStdout = True, ignoreStderr = True)
+
+
+def stop(name: str, networkName: str) -> None:
+    stopContainer(name)
+    removeNetwork(networkName)
+
+
+def manifestInspect(repository: str, tag: str) -> Dict[str, Any]:
+    _, output, _ = command(["docker", "manifest", "inspect", f"{repository}:{tag}", "--verbose"], ignoreStdout = True)
+    jsonOutput = json.loads(output)
+    if not isinstance(jsonOutput, dict):
+        raise TypeError(f"Invalid function result type \"{type(jsonOutput)}\". Expected: \"dict\"")
+
+    return jsonOutput
+
+
+def imageInspect(repository: str, tag: str) -> Dict[str, Any]:
+    _, output, _ = command(["docker", "image", "inspect", f"{repository}:{tag}"], ignoreStdout = True)
+    jsonOutput = json.loads(output[0]) # Since we are inspecting image with specific repository AND tag output will be a list with single object
```

Comment:
This will not work. `output` is of type `str` and indexing it with `0` index will give you the character at first position of that string